internal/engine/docker_compose_build_and_deployer.go

```diff
@@ -23,15 +28,15 @@ func NewDockerComposeBuildAndDeployer(dcc dockercompose.DockerComposeClient) *Do
 }
 
 // Extract the targets we can apply into DockerComposeTargets, or nil if we can't apply all targets.
-func (bd *DockerComposeBuildAndDeployer) extract(specs []model.TargetSpec) []model.DockerComposeTarget {
-	result := []model.DockerComposeTarget{}
+func (bd *DockerComposeBuildAndDeployer) extract(specs []model.TargetSpec) []targetTuple {
+	result := []targetTuple{}
 	for _, s := range specs {
 		dc, isDC := s.(model.DockerComposeTarget)
 		if !isDC {
 
 			return nil
 		}
-		result = append(result, dc)
+		result = append(result, targetTuple{s.ID(), dc})
```

Comment:
...doesn't a DockerComposeTarget (necessarily) implement `.ID()`? If so, this tuple situation is unnecessary.